talentprotocol/mvp
# Use noticed gem

## Summary

### What changed?

Notifications are now implemented using [Noticed](https://github.com/excid3/noticed) gem and the "Talent listed" notification was removed. From the user perspective, the only visible change is that the "Talent listed" notification doesn't exist anymore; everything else should be exactly the same.

### Why it changed?

Noticed includes functionality to easily send notifications by email (which we'll need) and also makes it easy to add new notification types.

### How it changed?

- Fields were added and removed from the `notifications` database table to match Noticed expect fields. The old fields were migrated to the new format.
- The existing Notification models were replaced by other classes deriving from Noticed class hierarchy.
- Notification text is now inferred from the notification type at display time, instead of being stored in the notification.

## Notion link

[Notification System](https://www.notion.so/talentprotocol/Notification-System-4a4d3446ca8444db80dcedbe46291f3e)

## How to test?

- Run the database migration
- Confirm that the existing notifications are working correctly
- Confirm that new notifications are created correctly

## Notes

The database migration makes backward incompatible changes to the schema, so the application should be put on maintenance mode during the deployment.
